pilot/pkg/xds/endpoints/endpoint_builder.go

```diff
@@ -602,7 +602,29 @@ func ExtractEnvoyEndpoints(locEps []*LocalityEndpoints) []*endpoint.LocalityLbEn
 
 // buildEnvoyLbEndpoint packs the endpoint based on istio info.
 func buildEnvoyLbEndpoint(b *EndpointBuilder, e *model.IstioEndpoint, mtlsEnabled bool) *endpoint.LbEndpoint {
-	addr := util.BuildAddress(e.Address, e.EndpointPort)
+	addr := util.BuildAddress(e.Addresses[0], e.EndpointPort)
+
+	// This change can support multiple addresses for an endpoint, then there are some use cases for it, such as
+	// 1. An endpoint can have both ipv4 or ipv6
+	// 2. An endpoint can be represented a serviceentry/workload instance with multiple IP addresses
+	// When the additional_addresses field is populated for EDS in Envoy configuration, there would be a Happy Eyeballs
```

Comment:
I'm not sure if "Happy Eyeballs" is the desired behavior, but that would be a separate Envoy improvement.
Assuming an IPv4 request is received, I would prefer it to be sent first to an address matching the request's IP address. If sending to an address with a matching IP family is not possible, then we would fall back to others.